### ISSUE TITLE
Infinite loop if 404 error while removing document (function deleteByMongoId)

### ISSUE THREAD

**Issue by adrianocola** [taterbase/mongoosastic#46]
The deleteByMongoId function is recursive and tries to remove a document a certain number o times, as specified by the parameter tries:

``` javascript
function deleteByMongoId(client, model,indexName, typeName, tries){
   ...
}
```

In case of an error the function calls itself to try again:

``` javascript
deleteByMongoId(client, model, indexName, typeName,tries--, cb);
```

The -- operator only subtracts 1 from tries after tries was passed again to the function, meaning that tries is always the same and the function will call itself forever.

Simply change <b>tries--</b> to <b>tries-1</b> and the problem is gone!

**Comment by taterbase** [comment]
This issue is so old and you probably don't care but a `--tries` ought to work as well right?

**Comment by taterbase** [comment]
Changes should be available in `0.2.6`
